src/Data/Singletons/Partition.hs

```diff
@@ -49,17 +49,23 @@ data PartitionedDecs =
         , pd_ty_syn_decs :: [TySynDecl]
         , pd_open_type_family_decs :: [OpenTypeFamilyDecl]
         , pd_closed_type_family_decs :: [ClosedTypeFamilyDecl]
+        , pd_standalone_kind_sigs :: Map Name DKind
```

Comment:
I'm split on this part of the patch. I added this in an effort to make reifying locally declared SAKS slightly more efficient. I'm not sure that the amount of code that we have to add to support this (see all the call sites of `dsReifyTypeWith`) is worth it. Perhaps effort would be better spent making local reification more efficient on `th-desugar`'s end? After all, locally reifying something in `th-desugar` basically amounts to scanning through a giant `[Dec]`.

Reply:
I've filed an issue about the time complexity of local reification in goldfirere/th-desugar#130. In the meantime, I'll do a quick microbenchmark and see how long it takes to compile `singletons` with and without the `pd_standalone_kind_sigs` hack. If the difference is negligible, I think I'll just rip out the hack.

Reply:
The results of my microbenchmark confirm my suspicions: the `pd_standalone_kind_sigs` hack isn't pulling its weight. Here is the time to compile `singletons` with the hack:

```
real    3m30.704s
user    3m23.989s
sys     0m2.987s
```

And without the hack:

```
real    3m27.500s
user    3m24.402s
sys     0m2.839s
```

I've ripped out the hack in my latest update. Let's pursue optimizing the speed of `dsReifyType` in goldfirere/th-desugar#130 instead.